### PR TITLE
Qualify external-attacher 2.1.1 in RC staging; enable Prometheus metrics in attacher, provisioner, and resizer

### PR DESCRIPTION
**Pull request by verult**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**: Qualify external-attacher 2.1.1 in RC staging; enable Prometheus metrics in attacher, provisioner, and resizer

```release-note
NONE
```

/assign @msau42

### CODE REVIEW

File: deploy/kubernetes/overlays/prow-gke-release-staging-rc/provisioner_metric_args.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+- op: add
+  path: /spec/template/spec/containers/0/args/-
```
Comment (msau42) [inline deploy/kubernetes/overlays/prow-gke-release-staging-rc/provisioner_metric_args.yaml]: Can we add comments saying which containers these applies to?

File: deploy/kubernetes/overlays/prow-gke-release-staging-rc/provisioner_metric_args.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  value: "--metrics-address=:8080"
```
Comment (msau42) [inline deploy/kubernetes/overlays/prow-gke-release-staging-rc/provisioner_metric_args.yaml]: Will all of the sidecars listening on 8080 conflict? Also will probably cause issues with hostNetwork as well

**Comment by verult** [comment]
Addressed comments!

File: deploy/kubernetes/overlays/prow-gke-release-staging-rc/kustomization.yaml [hunk 3]
```diff
@@ -11,10 +11,18 @@ images:
...
   newName: gke.gcr.io/gke-release-staging/csi-resizer
   newTag: "v0.4.0-gke.0"
+
+patchesJson6902:
+- target:
+    group: apps
+    version: v1
+    kind: StatefulSet
+    name: csi-gce-pd-controller
+  path: provisioner_metric_args.yaml
```
Comment (msau42) [inline deploy/kubernetes/overlays/prow-gke-release-staging-rc/kustomization.yaml]: a more fitting file name could be "enable_sidecar_metrics"

Comment (verult) [reply deploy/kubernetes/overlays/prow-gke-release-staging-rc/kustomization.yaml]: ty for the catch!

**Pull request merged**